### PR TITLE
[#52] drain.sh scaledown should wait for broker to stop and verify messageCount==0 before exit success

### LINKED ISSUE #52: drain.sh scaledown should wait for broker to stop and verify messageCount==0 before exit success
**Describe the bug**
The drain.sh in amq-launch-module should wait for the broker (which is running background) to finish
before exit. This is useful for broker to shutdown properly and for checking out broker status when scale
down fails.

**To Reproduce**
1. Get X version of the source
2. '...'

**Platform (please complete the following information):**
 - OS: [e.g. Fedora]
 - Version [e.g. 32]
 - CEKit version [e.g. 3.5.0]
 - Container build tool [e.g. podman]

**Expected behavior**
Optional, use if not obvious. A clear and concise description of what you expected to happen.

**Screenshots**
Optional, add screenshots to help explain your problem.

**Additional context**
Optional. Add any other context about the problem here.

### PR DESCRIPTION
**Pull request by gaohoward**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

**Comment by gaohoward** [comment]
I did some manual test to make sure the script works.

File: modules/activemq-artemis-launch/added/drain.sh [hunk 1]
```diff
@@ -134,5 +134,35 @@ if [ "${HTTP_CODE}" != "200" ]; then
   exit 1
 fi
 
+#restart the broker to check messages
```
Comment (gtully) [inline modules/activemq-artemis-launch/added/drain.sh:137]: In normal circumstances the broker will stop but do we need to wait for stopped?

I am wondering if there is a possibility that the restart fails if the broker is still stopping

Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh:137]: The restart command actually do a stop and start. The stop will wait for the process to end. It's like this
```
stop() {
  if [ -f "${PID_FILE}" ] ; then
    pid=`cat "${PID_FILE}"`
    kill $@ ${pid} > /dev/null
  fi
  for i in 1 2 3 4 5 ; do
    status
    if [ $? -ne 0 ] ; then
      return 0
    fi
    sleep 1
  done
  echo "Could not stop process ${pid}"
  return 1
}
```

Comment (gtully) [reply modules/activemq-artemis-launch/added/drain.sh:137]: ok, it gives it 5 seconds, i hope that is enough in all normal circumstances, I wonder should it do a kill -9 as it has sent a TERM, telling it to gracefully stop.

**Review by gtully — COMMENTED** [review]
this is a good improvement in any event.

File: modules/activemq-artemis-launch/added/drain.sh (hunk 1 above)
Comment (gtully) [reply modules/activemq-artemis-launch/added/drain.sh:137]: there is a force-stop, https://github.com/apache/activemq-artemis/blob/cd1e9822f597b33474883c3c09397b0a7ef82f8c/artemis-cli/src/main/resources/org/apache/activemq/artemis/cli/commands/bin/artemis-service#L124

maybe we need to use stop, and if it fails use force-stop, then start. (as in, have have to do our own restart logic) or maybe fix the restart script.

Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh:137]: makes sense. I'll use force stop. Thanks!

Comment (gaohoward) [reply modules/activemq-artemis-launch/added/drain.sh:137]: @gtully Updated the branch. Pls review.

**Review by gtully — APPROVED** [review]

**Review by brusdev — APPROVED** [review]

**Pull request merged**